CPAC/connectome/connectivity_matrix.py

```diff
@@ -0,0 +1,274 @@
+from CPAC.utils.datasource import create_roi_mask_dataflow
+from CPAC.utils.interfaces.function import Function
+from CPAC.utils.interfaces.netcorr import NetCorr
+
+logger = logging.getLogger('nipype.workflow')
+methods = {
+    'AFNI': {'Pearson': '', 'Partial': '-part_corr'},
+    'Nilearn': {'Pearson': 'correlation', 'Partial': 'partial correlation'}
+}
+tools = {'afni': 'AFNI', 'nilearn': 'Nilearn'}
+
+
+def _connectome_name(time_series, method):
+    """Helper function to create connectome file filename
+
+    Parameters
+    ----------
+    time_series : str
+        path to input time series
+
+    method : str
+        BIDS entity value for `desc-` key
+
+    Returns
+    -------
+    str
+    """
+    method = method.replace(' ', '+')
+    return os.path.abspath(os.path.join(
+        os.path.dirname(time_series),
+        time_series.replace(
+            '_timeseries.1D',
+            f'_desc-{method}_connectome.npy'
+        )
+    ))
+
+
+def _get_method(method, tool):
+    """Helper function to get tool's method string
+
+    Parameters
+    ----------
+    method : str
+
+    tool : str
+
+    Returns
+    -------
+    str or NotImplemented
+    """
+    tool = tools[tool.lower()]  # make case-insensitive
+    cm_method = methods[tool].get(method, NotImplemented)
+    if cm_method is NotImplemented:
+        logger.warning(NotImplementedError(
+            f'{method} has not yet been implemented for {tool} in C-PAC.'
+        ))
+    return cm_method
+
+
+def compute_correlation(time_series, method):
+    """Function to create a numpy array file [1] containing a
+    correlation matrix for a given time series and method.
+
+    Parameters
+    ----------
+    time_series : str
+        path to time series output
+
+    method : str
+        correlation matrix method. See https://fcp-indi.github.io/docs/nightly/user/tse#configuring-roi-time-series-extraction
+        for options and how to configure.
+
+    References
+    ----------
+    .. [1] The NumPy community (2021). NPY format. https://numpy.org/doc/stable/reference/generated/numpy.lib.format.html#npy-format
+    """  # pylint: disable=line-too-long  # noqa E501
+    desc_regex = re.compile(r'\_desc.*(?=\_)')
+
+    data = np.genfromtxt(time_series).T
+
+    existing_desc = desc_regex.search(time_series)
+    if hasattr(existing_desc, 'group'):
+        matrix_file = time_series.replace(
+            existing_desc.group(),
+            '+'.join([
+                existing_desc.group(),
+                method
+            ])
+        ).replace('_timeseries.1D', '_connectome.npy')
+    else:
+        matrix_file = _connectome_name(time_series, method)
+
+    method = _get_method(method, 'Nilearn')
+    if method is NotImplemented:
+        return NotImplemented
+
+    connectivity_measure = ConnectivityMeasure(kind=method)
+    connectome = connectivity_measure.fit_transform([data])[0]
+
+    np.save(matrix_file, connectome)
+    return matrix_file
```

Comment:
I tried to adapt this from what was already in C-PAC: https://github.com/FCP-INDI/C-PAC/blob/fc5971a0a4d9a800dd353a81de58d89de673ac3f/CPAC/connectome/pipeline.py#L16-L30